### ISSUE TITLE
AttributeError: 'list' object has no attribute 'lower'

### ISSUE THREAD

**Issue by trillex** [Diaoul/subliminal#761]
I seem to consistently get the following error, when running Subliminal against my movies or series, using pretty much just a standard run (subliminal download -l en -l da /path/to). 

Here is the output: 

```
Downloading subtitles  [#####################---------------]   59%  0d 02:48:08  Two and a Half Men - S04E02 - Who's Vod Kanockers - 720p WEB-DL.mkv
Traceback (most recent call last):
  File "/volume1/@appstore/py3k/usr/local/bin/subliminal", line 11, in <module>
    sys.exit(subliminal())
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/cli.py", line 403, in download
    hearing_impaired=hearing_impaired, only_one=single)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/core.py", line 211, in download_best_subtitles
    for s in subtitles], key=operator.itemgetter(1), reverse=True)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/core.py", line 211, in <listcomp>
    for s in subtitles], key=operator.itemgetter(1), reverse=True)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/score.py", line 106, in compute_score
    matches = subtitle.get_matches(video)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/providers/opensubtitles.py", line 78, in get_matches
    matches |= guess_matches(video, guessit(self.movie_release_name, {'type': 'episode'}))
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/subliminal/subtitle.py", line 235, in guess_matches
    if video.format and 'format' in guess and not isinstance(guess['format'], list) and guess['format'].lower() == video.format.lower():
AttributeError: 'list' object has no attribute 'lower'
```

Seems to be related to having two formats attached to it. The best I have been able to google myself to is: https://github.com/pymedusa/Medusa/issues/1033, which fixed the issue by changing code in their software and not in subliminal itself. 

I have attached a debug with one of the files that I can reproduce it with without fail. 

[debug.txt](https://github.com/Diaoul/subliminal/files/1050801/debug.txt)

Very surprised no one else have run into this issue.

**Comment by ratoaq2** [comment]
guessit (the library used to parse release names) can potentially return multiple values for several fields. Some fields are usually single value, but because of wrong parsing/guess or because of some uncommon scenario the result might be multiple values. E.g.: 
- you could get `BluRay` and `WEB-DL` in `format` field when parsing a release name (`abc.WEB.BD.1080p`) which refers to a pack of episodes. Guessit returns correctly multiple values. 
- you could wrongly get `BluRay` and `WEB-DL` for a strange release `abc s01e02 Strange Things on Web BLURAY 1080p`. Guessit wrongly returns multiple values. 

guessit parser is powerful and it is always improving, but it's not perfect. 

Since guessit could detect multiple values in some cases for certain properties, Subliminal needs to handle that. 

The other software you mentioned fixed the issue by adding an extra custom rule to guessit to discard and log multiple values on the `format` property. That way, guessit will never return more than one value for that property and this exception will never happen in subliminal. So, it's a workaround.

**Comment by ratoaq2** [comment]
In your specific case, given your filename (`Two and a Half Men - S04E02 - Who's Vod Kanockers - 720p WEB-DL.mkv`) guessit will return `VOD` (Video on Demand) and `WEB-DL` as `format`

**Comment by Diaoul** [comment]
@ratoaq2 any suggesting on how to fix this?

**Comment by ratoaq2** [comment]
In short, just drop both `lower` calls in this line:
https://github.com/Diaoul/subliminal/blob/master/subliminal/subtitle.py#L235

`format` has a well defined list of possible values. It's not a free text. 

The rest of the fields looks fine: guessit will only return 1 release group, 1 title (there was a bug in earlier versions). 
I'll need to check the episode_title.

**Comment by fishmongr** [comment]
I just ran into this as well. Instead of removing the lower() comparisons I created a pull request to just check if the value is of type string before calling #800

**Comment by saxicek** [comment]
Fix by @sprt (https://github.com/sprt/subliminal/commit/f4f51a088f37f1acf7cb4f4972c2ae6e16def1a6) worked for me.

**Comment by CWempe** [comment]
An change there will be a PR to fix this issue?
@sprt

I am using the official docker image (https://hub.docker.com/r/diaoulael/subliminal/) wich make it complicate to manually fix this issue.

**Comment by sprt** [comment]
@CWempe Be advised the patch I suggested doesn't appear to be correct; see https://github.com/Diaoul/subliminal/pull/769#discussion_r151983504. Nonetheless, for your problem, I'd say apply a patch locally and build a new Docker container.

**Comment by CWempe** [comment]
Thanks.

I build my own docker image based on the `feature/video_format` branch.  
https://github.com/Diaoul/subliminal/tree/feature/video_format

Seems to work. 😄 

If anybody is interessted: https://hub.docker.com/r/cwempe/subliminal/

**Comment by ratoaq2** [comment]
Fixed by 3121a75499779c7cc9eba059a4fa123cf5541027
